flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -685,13 +691,24 @@ JobManagerRunner createJobManagerRunner(JobGraph jobGraph, long initializationTi
 
     @Override
     public CompletableFuture<String> triggerSavepoint(
-            final JobID jobId,
+            final AsynchronousJobOperationKey operationKey,
```

Comment:
The AsynchronousJobOperationKey could be an implementation detail of the Dispatcher/cache; to prevent it from leaking into the REST API.

Reply:
Hm, I saw that the operation key is created in `AbstractAsynchronousOperationHandlers`, so I thought this would be the natural place to start using it.